### ISSUE TITLE
Add basic querying support to the library

### ISSUE THREAD

**Issue by SubiyaCryolite** [SubiyaCryolite/Jenesis-Data-Store#1]
Add basic querying support to the library. Use this to load types based on the values of mapped fields.

**Comment by SubiyaCryolite** [comment]
This is a WIP. Should be in next stable release.

**Comment by SubiyaCryolite** [comment]
Be sure that all of these operators are supported
```xml
=
!=
<>
>
<
>=
<=
!<
!>
ALL
AND
ANY
BETWEEN
EXISTS
IN
LIKE
NOT
OR
IS NULL
UNIQUE
```

**Comment by SubiyaCryolite** [comment]
Closed with release v1.170626
